docs/readability/not_using_items_to_iterate_over_a_dictionary.rst

```diff
@@ -18,19 +18,30 @@ The code below defines a for loop that iterates over a dictionary named ``d``. F
 Best-practice
 -------------
 
-Use ``items()`` to iterate across dictionary
-............................................
+Use ``iteritems()`` to iterate across dictionary
+................................................
 
-The updated code below demonstrates the Pythonic style for iterating through a dictionary. When you define two variables in a ``for`` loop in conjunction with a call to ``items()`` on a dictionary, Python automatically assigns the first variable as the name of a key in that dictionary, and the second variable as the corresponding value for that key.
+The updated code below demonstrates the Pythonic style for iterating through a dictionary. When you define two variables in a ``for`` loop in conjunction with a call to ``iteritems()`` on a dictionary, Python automatically assigns the first variable as the name of a key in that dictionary, and the second variable as the corresponding value for that key.
 
 .. code:: python
 
     d = {"first_name": "Alfred", "last_name":"Hitchcock"}
     
-    for key,val in d.items():
+    for key,val in d.iteritems():
         print "{} = {}".format(key, val)
 
+``items()`` may be another choice to iterate a dictionary, but in fact, ``iteritems()`` is better.
+
+In Python 2, Python ``items()`` built a real list of tuples and returned that. That could potentially take a lot of extra memory.
+
+Then, generators were introduced to the language in general, and that method was reimplemented as a iterator-generator method named ``iteritems()``.
+
+In Python 3,the ``iteritems()`` is gone, instead, the ``items()`` returns an iterator just like how ``iteritems()`` works in Python 2.
```

Comment:
This is actually *not really true*. In python 3 `items` returns a `view object` to the items and not an iterator like the `iteritems` in python 2.7. 

See: 
https://docs.python.org/2/library/stdtypes.html#dict.iteritems
https://docs.python.org/3.3/library/stdtypes.html#dict.items